File: uchat-mvp/test_code_example.py
"""
Example Python code for testing Code Preview feature
"""
import os
import sys
from typing import List, Optional


class CodePreviewDemo:
    """Demo class for code preview testing."""
    
    def __init__(self, name: str):
        self.name = name
        self.items: List[str] = []
    
    def add_item(self, item: str) -> None:
        """Add an item to the list."""
        if item:
            self.items.append(item)
            print(f"Added: {item}")
        else:
            raise ValueError("Item cannot be empty")
    
    def get_items(self) -> List[str]:
        """Return all items."""
        return self.items.copy()
    
    def process_data(self, data: dict) -> Optional[str]:
        """Process some data and return result."""
        if not data:
            return None
        
        result = []
        for key, value in data.items():
            if isinstance(value, (int, float)):
                result.append(f"{key}: {value * 2}")
            else:
                result.append(f"{key}: {value}")
        
        return "\n".join(result)


def main():
    """Main function to demonstrate the code."""
    demo = CodePreviewDemo("Test")
    
    # Add some items
    demo.add_item("Python")
    demo.add_item("JavaScript")
    demo.add_item("C#")
    
    # Process data
    data = {
        "count": 42,
        "name": "example",
        "ratio": 3.14
    }
    
    result = demo.process_data(data)
    print(result)
    
    # Get all items
    items = demo.get_items()
    print(f"Total items: {len(items)}")


if __name__ == "__main__":
    main()

File: uchat-mvp/test_code_example.js
/**
 * Example JavaScript code for testing Code Preview feature
 */

class CodePreviewDemo {
    /**
     * Creates a new CodePreviewDemo instance
     * @param {string} name - The name of the demo
     */
    constructor(name) {
        this.name = name;
        this.items = [];
    }

    /**
     * Adds an item to the collection
     * @param {string} item - The item to add
     * @throws {Error} If item is empty
     */
    addItem(item) {
        if (!item) {
            throw new Error('Item cannot be empty');
        }
        
        this.items.push(item);
        console.log(`Added: ${item}`);
    }

    /**
     * Gets all items in the collection
     * @returns {Array<string>} A copy of the items array
     */
    getItems() {
        return [...this.items];
    }

    /**
     * Processes data and returns formatted result
     * @param {Object} data - The data to process
     * @returns {string} The formatted result
     */
    processData(data) {
        if (!data || Object.keys(data).length === 0) {
            return '';
        }

        const results = [];

        for (const [key, value] of Object.entries(data)) {
            if (typeof value === 'number') {
                results.push(`${key}: ${value * 2}`);
            } else {
                results.push(`${key}: ${value}`);
            }
        }

        return results.join('\n');
    }

    /**
     * Gets the count of items
     * @returns {number} The number of items
     */
    get itemCount() {
        return this.items.length;
    }
}

/**
 * Main function to demonstrate the code
 */
function main() {
    const demo = new CodePreviewDemo('Test');

    // Add some items
    demo.addItem('JavaScript');
    demo.addItem('Python');
    demo.addItem('C#');

    // Process data
    const data = {
        count: 42,
        name: 'example',
        ratio: 3.14
    };

    const result = demo.processData(data);
    console.log(result);

    // Display item count
    console.log(`Total items: ${demo.itemCount}`);
}

// Run the main function
if (require.main === module) {
    main();
}

module.exports = CodePreviewDemo;
